dokka-runners/dokka-gradle-plugin/build.gradle.kts

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright 2014-2024 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+@file:Suppress("UnstableApiUsage") // jvm test suites
+
+import dokkabuild.tasks.GenerateDokkaGradlePluginConstants
+import dokkabuild.utils.skipTestFixturesPublications
+
+plugins {
+    id("dokkabuild.gradle-plugin")
+
+    kotlin("plugin.serialization") version embeddedKotlinVersion
+
+    id("dokkabuild.dev-maven-publish")
+    alias(libs.plugins.kotlinx.binaryCompatibilityValidator)
+    `java-test-fixtures`
+    `jvm-test-suite`
+
+    id("dokkabuild.publish-base")
+    alias(libs.plugins.gradlePluginPublish)
+}
+
+description = "Gradle plugin for using Dokka Engine"
+
+kotlin {
+    compilerOptions {
+        optIn.addAll(
+            "kotlin.RequiresOptIn",
+            "org.jetbrains.dokka.gradle.internal.DokkaInternalApi",
+            "kotlin.io.path.ExperimentalPathApi",
```

Comment:
minor: how hard it would be not to depend on experimental stdlib API?

Reply:
At the moment it's the experimental path functions are only used in test sources. It's a bit tricky/ugly to only opt-in for non-main sources.